[Manh Duc] iP

hmanhduc2k opened the pull request:
# DuC (That time I create a chatbot with my name for CS2103T IP because Singaporeans keep mispronouncing it then asking me how  :joy::joy::thinking::vomiting_face:)

Yea you heard me right. The most generic yet difficult-to-pronounce chatbot you will have ever used. Why use me then?
Because I am **generic**, it means that there is absolutely no risk of using me. I can be the most _useful~~~_ chat bot you will ever use without even notice. Because I'm doing my job properly and not trying to market you with distrustful stuffs.

Most importantly, please learn how to pronounce my name after using DuC. Concluded with thanks.

So, enough ~~hidden marketing~~ long-winding introduction that sound like a rip-off light novel title already. What can I offer to you?

1. Schedule a to-do task
2. Schedule a deadline task with prescribed date
3. Schedule an event task with occurred date
4. Track task or delete task on completion - of your choice
5. Find task by date or by keywords

Does that sound too generic? I also offer this:
* Easy to use and remember command line
* Difficult to pronounce names that will make you question how to pronounce it 100% of the time
* Some other things because CS2103T requires so

How does DuC sound like to you? Here is a sample list of daily schedule you can follow with DuC
- [x] [T] code
- [ ] [D] code while sleeping (by everyday)
- [ ] [T] code while eating
- [ ] [T] code while taking a sh*t
- [ ] [E] code while crying in the corner of your dorm room because ~~you have no social life and are a miserable victim of crony capitalism and materialism~~ you cannot contain excessive joy in doing programming

How about a glimpse of my development process?
``` java
case "todo":
      return new TaskCommand(description, taskList, Task.Type.TODO);
case "deadline":
      return new TaskCommand(description, taskList, Task.Type.DEADLINE);
case "event":
      return new TaskCommand(description, taskList, Task.Type.EVENT);
```

Basically, there are three types of task you can add to your list:
`<todo>`,`<deadline>`,`<event>`. Sounds very generic is it? I know

Here is a random filler quote because I don't know what else to add:
> Death is like the wind. Always by my side - Yasuo

To conclude, please click [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ) if you are interested to find out more about my project. 

Nah I'm just messing with you, do click [here](https://github.com/hmanhduc2k/ip/releases/download/v01/DuC-1.0-all.jar) instead to try out my product.

Peace~~~

Commit:
Add Gradle support

hmanhduc2k committed:
test commit functions

hmanhduc2k committed:
Revert "test commit functions"

This reverts commit a8430b0f

hmanhduc2k committed:
Test commit functions 2

hmanhduc2k committed:
Duke Level 1: Greet, Echo, Exit

hmanhduc2k committed:
Duke Level 2: Add, List

hmanhduc2k committed:
Duke Level 3: Mark as Done

hmanhduc2k committed:
Duke Level-4: ToDo, Event, Deadline (First attempt)

hmanhduc2k committed:
Duke Level-4: ToDo, Event, Deadline (Second attempt)

hmanhduc2k committed:
Duke A-TextUiTesting: Automated Text UI Testing

hmanhduc2k committed:
Duke Level-5: Handle Errors

hmanhduc2k committed:
Duke Level-6: Delete

hmanhduc2k committed:
Duke A-Enums: Use Enums if-applicable

hmanhduc2k committed:
Minor changes in arrangement of classes and methods

hmanhduc2k committed:
Changes of class and method structuring for all levels

hmanhduc2k committed:
Minor code cleanup

hmanhduc2k committed:
Create New Branch

hmanhduc2k committed:
Duke Level-7: Save

hmanhduc2k committed:
Duke Level-7: Save

hmanhduc2k committed:
Duke Level-8: Date and Time (for Date only)

hmanhduc2k committed:
Duke A-MoreOOP: Attempt 1

hmanhduc2k committed:
Duke A-MoreOOP: Attempt 2 (Functioning with all classes separated)

hmanhduc2k committed:
temporary changes to test branch merging

hmanhduc2k committed:
Merge branch 'master' into branch-Level-7

# Conflicts:
#	src/main/java/Duke/DukeException.java

hmanhduc2k committed:
Change Duke Exception error message content for invalid date format

hmanhduc2k committed:
Delete Command Package

hmanhduc2k committed:
Duke A-Packages: Allocate classes into packages

hmanhduc2k committed:
Add Command packages and classes

hmanhduc2k committed:
Merge branch 'A-Packages'

# Conflicts:
#	src/main/java/Duke/Command/Command.java
#	src/main/java/Duke/Main/Parser.java

hmanhduc2k committed:
Duke A-JavaDoc: Add Java Doc header and class explanation to Duke

hmanhduc2k committed:
Duke A-CodingStandard: Follow the prescribed coding standard set by CS2103T guideline

hmanhduc2k committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/Duke/Main/Duke.java
#	src/main/java/Duke/Main/DukeException.java

hmanhduc2k committed:
Edit enum naming for Duke Exception type

hmanhduc2k committed:
Duke Level-9: Find (functional with query search works)

hmanhduc2k committed:
Merge branch 'master' into branch-Level-7

hmanhduc2k committed:
Adding Duke Exception Class

hmanhduc2k committed:
Changing find command structure

hmanhduc2k committed:
Changing access modifier to Parser in java and restructure exception package for later OOP

hmanhduc2k committed:
Merge branch 'branch-Level-9'

hmanhduc2k committed:
Duke A-JUnit: Test cases are added for Parser class to test control flow mechanism

hmanhduc2k committed:
Duke A-Jar: Jar file created and stored on the Artifacts directory

hmanhduc2k committed:
Merge branch 'branch-A-JUnit'

hmanhduc2k committed:
Duke A-MoreOOP: Functional with Command classes being added, exception classes distinguished, and other components fully specified

hmanhduc2k committed:
Change test cases for Duke

hmanhduc2k committed:
Merge branch 'branch-A-MoreOOP'

# Conflicts:
#	src/test/java/Duke/Main/ParserTest.java

hmanhduc2k committed:
Change command statement

hmanhduc2k committed:
Change command statement

hmanhduc2k committed:
More JUnit Test

hmanhduc2k committed:
Update logo for DuC

hmanhduc2k committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-add-gradle

hmanhduc2k committed:
add and merge gradle branch successfully

hmanhduc2k committed:
Merge branch 'branch-add-gradle'

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

hmanhduc2k committed:
Debug gradle file and config build for running DuC

hmanhduc2k committed:
Debug gradle files, add more dependencies and create executable JAR file with shadowJar

hmanhduc2k committed:
Add checkstyle.xml, config Intellij and other areas so that checkstyle tools can be used

hmanhduc2k committed:
Change README.md to suit individual project

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Update README.md to include JAR file release

dannytayjy commented on src/main/java/Duke/Command/DateOfCommand.java:
```diff
@@ -0,0 +1,40 @@
+public class DateOfCommand extends Command {
+
+    private String description;
+    private TaskList taskList;
+    public DateOfCommand(String description, TaskList taskList) {
```
Missing bank line separation?

dannytayjy commented on src/main/java/Duke/Command/DateOfCommand.java:
```diff
@@ -0,0 +1,40 @@
+public class DateOfCommand extends Command {
+
+    private String description;
+    private TaskList taskList;
+    public DateOfCommand(String description, TaskList taskList) {
```
Having a bank line separation helps in enhancing readability

dannytayjy replied:
I noticed the same issue in several other places too

dannytayjy replied:
Having a bank line separation between the variable and the constructor helps in enhancing readability. I noticed the same issue in several other places too.

dannytayjy commented on src/main/java/Duke/Command/DateOfCommand.java:
```diff
@@ -0,0 +1,40 @@
+            throw new DukeIncompleteException();
+        }
+        try {
+            int index = Integer.parseInt(description);
+            Task task = taskList.get(index);
+            String type = (task instanceof Deadline) ? "Deadline" :
+                    (task instanceof Event) ? "Event" : "Todo";
+            if (type.equals("Todo")) {
```
Having a bank line separation helps in enhancing readability

hmanhduc2k replied:
Thanks for the suggestion, I will consider adding a blank line :)

dannytayjy commented on src/main/java/Duke/Command/DateOfCommand.java:
```diff
@@ -0,0 +1,40 @@
+            if (type.equals("Todo")) {
+                return task.getDate() + index;
+            } else {
+                return "The " + type + " time of task " +
```
The "+" sign should be at the next line as per coding standard

dannytayjy commented on src/main/java/Duke/Command/DateOfCommand.java:
```diff
@@ -0,0 +1,40 @@
+            if (type.equals("Todo")) {
+                return task.getDate() + index;
+            } else {
+                return "The " + type + " time of task " +
```
The "+" sign at the end should be at the next line as per coding standard

dannytayjy replied:
I noticed the same issue in several other places too

hmanhduc2k replied:
Noted, with thanks

dannytayjy commented on src/main/java/Duke/Command/DateOfCommand.java:
```diff
@@ -0,0 +1,40 @@
+    public String reply() {
+        if (description.length() == 0) {
+            throw new DukeIncompleteException();
+        }
+        try {
```
Logical units within a block should be separated by one blank line. I noticed the same issue in several other places too.

dannytayjy commented on src/main/java/Duke/Command/FindCommand.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public String reply() {
+        StringBuilder output = new StringBuilder("The following task(s) matches query '" +
+                description + "':\n");
+        int count = 0;
+        for (Task currentTask : taskList) {
+            if (currentTask.getTaskName().contains(description)) {
+                output.append(count + 1).append(". ").append(currentTask).append("\n");
+                count++;
+            }
+        }
+        if (count == 0) {
```
Having bank line separation between different blocks helps in enhancing readability

dannytayjy reviewed (commented):
Your code quality seems okay to me but I may have missed out on certain things as it was getting hard to read without blank line separation between different blocks.

It would be good to have blank line separation as it helps in readability.

YaleChen299 commented on src/main/java/Duke/DukeException/DukeException.java:
```diff
@@ -0,0 +1,36 @@
+
+    /**
+     * Constructor for Duke Exception class
+     * @param message Error message
```
Maybe adding a line after the brief description before the parameter section would help to standardize the java doc. This applies to other places too.

hmanhduc2k replied:
Thanks for your suggestion, I will try to change how my JavaDoc is like

hmanhduc2k committed:
Add GUI with functional features, rename some files to match with coding standard set by the module.

hmanhduc2k committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	README.md

YaleChen299 commented on src/main/java/Duke/Main/Ui.java:
```diff
@@ -0,0 +1,63 @@
+    private final Scanner sc;
+
+    /**
+     * Constructor for the UI class, setting up a Scanner object
```
According to the standard in this course, it is better to add a full stop at the end of the brief description.

YaleChen299 reviewed (commented):
Generally having good coding quality. However, maybe you can improve by adding more java doc and standardizing the them with the course coding standard. 😉

hmanhduc2k committed:
Merging GUI to Duke and rearrange codebase

hmanhduc2k committed:
Merging GUI to Duke and rearrange codebase

hmanhduc2k committed:
Add varargs methods to TaskList.java

hmanhduc2k committed:
Modify codebase to match with CS2103T coding standard, add more JavaDoc comments to methods, customizing GUI and add waiting time between each input-reply cycle

hmanhduc2k committed:
Customize GUI for Duke

Set the label text box to change its height based on the text size input, so as to wrap the conversation completely and avoid cutting text.

Make the Duke window stage not resizable to avoid detached components being enlarged.

Increase the width of Duke's GUI components for ease of usage.

Remove redundant import lines to match coding standard.

hmanhduc2k committed:
Add assertions to codebase and modify minor details

Add assertions at some breakpoints in Duke, in confirming the properties of GUI and reply feature. Assertions are added to document important aspect of the code.

Add background color to Anchor Pane for personalization

Remove redundant methods, imports, and comments to match coding standard

hmanhduc2k committed:
Improve code quality for Duke

Critically examine, assess, and modify codebase for Duke to match with the given quality standard as set by CS2103T.

Improve readability of code, rename variables to include their roles, scale-down deep-nested codes and complicated expressions, remove magic literals and put them as static constant instead.

Remove redundant methods and statements, remove exception enum.

hmanhduc2k committed:
Merge pull request #1 from hmanhduc2k/!branch-A-Assertions

Add assertions to codebase and modify minor details

hmanhduc2k committed:
Merge pull request #2 from hmanhduc2k/branch-A-CodeQuality

Improve code quality for Duke

hmanhduc2k committed:
Add stream to Java

Use and test out the Stream data structure of Java.

hmanhduc2k committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java

hmanhduc2k committed:
Create gradle.yml and add Continuous Integration

Set up CI for the project

hmanhduc2k committed:
Merge pull request #3 from hmanhduc2k/hmanhduc2k-patch-1

Create gradle.yml and add Continuous Integration

hmanhduc2k committed:
Add a new command called 'update' to Duke

The new command is part of the BCD extension, of which our team chooses category C. It is supposed to update/replace a task at the specified index, and is matched with C-Update.

The command formula is: update + (index number of task to be updated or replaced) + (task type) + (task description).

hmanhduc2k committed:
Merge pull request #4 from hmanhduc2k/branch-updateCommand

Add a new command called 'update' to Duke

hmanhduc2k committed:
Add local unit test cases to all Task classes

Add unit test cases to assert work functionality of various features and methods in Task, Todo, Deadline, and Event classes and objects. In particular, the test cases test whether the Task classes throw the correct exception, save the correct timing, transform the correct date format and save at the correct format, change status when mark as completed, and represent itself correctly as a string.

hmanhduc2k committed:
Merge pull request #6 from hmanhduc2k/add-more-test

Add local unit test cases to all Task classes

hmanhduc2k committed:
Improve GUI and user interaction

Improve GUI by adding background and customizing chat features

hmanhduc2k committed:
Improve GUI and user interaction

Improve GUI by adding background and customizing chat features, separating profile picture and label dialog

hmanhduc2k committed:
Minor fixes

hmanhduc2k committed:
Update User Guide for DuC

Update User Guide to detail all command and its formula

hmanhduc2k committed:
Minor fixes

hmanhduc2k committed:
Add Representative Image to product

hmanhduc2k committed:
Customization of DuC

Make everything more consistent throughout the product.

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Update README.md

Less profanity and Formalize User Guide

hmanhduc2k committed:
Update README.md

Replacing space character

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Create README.md

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Final commit to DuC IP

hmanhduc2k committed:
Final commit to DuC IP

Standardize all variables and JavaDoc header comments.

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Update README.md

hmanhduc2k committed:
Minor bug fixes and file path specifications

hmanhduc2k committed:
Merge remote-tracking branch 'origin/master'
